### DISCUSSION TITLE
React Native application module

### DISCUSSION THREAD

**Discussion by mkleszcz** [apptension/saas-boilerplate#335] [Ideas]
This is a proposition of implementing React Native application module for the SaaS Boilerplate.

### Module specification

- The application should be a part of the repository the same way as the current `webapp` module
- There should be a possibility to share some parts of the code between regular web application and mobile application, like for example using shared `webapp-api-client` package to communicate with the backend
- Mobile application should have implemented the same functionality as the web application like: user authorization, subscriptions, payments, CMS integration etc

**Comment by camstuart** [comment]
This would be extremely cool. I would suggest also:

- Expo is used, so that CI/CD can be done for the mobile app on the Expo platform
- Subscriptions and payments should either utilise revenue cat or be native to the mobile platform (ensuring compliance with Apple, and therefore increasing the chance of app store approval) 

Regarding the sharing of react components: From what I have read, this is tricky to do, and might be better as a 'stretch target'. Essentially the components must be pure, and have styling injected by the outer architecture. So having a location somewhere in `libs` that holds 'cross platform' components might be a possible approach.

My SaaS plan is to use this boilerplate and create a mobile app that talks to the gql endpoint in either react native or flutter.
So perhaps a good starting point is a guide for working with the backend in a secure manner from other technologies.
